### ISSUE TITLE
Unable to set mp_pose in my Apple Silicon Mac

### ISSUE THREAD

**Issue by rickypang0219** [google/mediapipe#5177]
# MacOS Version 
Sonoma 14.2.1 

# Software Version
```
python: 3.11.5
opencv-python  : 4.6.0
mediapip: 0.10.10
```

# Problem
I am a newbie in computer vision and motion detector. I am building an motion detection app in Python and I have to set the pose using mediapipe 

```python 
# Set up MediaPipe instance
mp_pose = mp.solutions.pose
pose = mp_pose.Pose(min_detection_confidence=0.5, min_tracking_confidence=0.5)
```

However, I got an error regarding initialising `pose` as below 

```---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[3], line 1
----> 1 pose = mp_pose.Pose(min_detection_confidence=0.5, min_tracking_confidence=0.5)

File /opt/homebrew/Caskroom/miniconda/base/lib/python3.11/site-packages/mediapipe/python/solutions/pose.py:146, in Pose.__init__(self, static_image_mode, model_complexity, smooth_landmarks, enable_segmentation, smooth_segmentation, min_detection_confidence, min_tracking_confidence)
    122 """Initializes a MediaPipe Pose object.
    123
    124 Args:
   (...)
    143     https://solutions.mediapipe.dev/pose#min_tracking_confidence.
    144 """
    145 _download_oss_pose_landmark_model(model_complexity)
--> 146 super().__init__(
    147     binary_graph_path=_BINARYPB_FILE_PATH,
    148     side_inputs={
    149         'model_complexity': model_complexity,
    150         'smooth_landmarks': smooth_landmarks and not static_image_mode,
    151         'enable_segmentation': enable_segmentation,
    152         'smooth_segmentation':
    153             smooth_segmentation and not static_image_mode,
    154         'use_prev_landmarks': not static_image_mode,
    155     },
    156     calculator_params={
    157         'posedetectioncpu__TensorsToDetectionsCalculator.min_score_thresh':
    158             min_detection_confidence,
    159         'poselandmarkbyroicpu__tensorstoposelandmarksandsegmentation__ThresholdingCalculator.threshold':
    160             min_tracking_confidence,
    161     },
    162     outputs=['pose_landmarks', 'pose_world_landmarks', 'segmentation_mask'])

File /opt/homebrew/Caskroom/miniconda/base/lib/python3.11/site-packages/mediapipe/python/solution_base.py:248, in SolutionBase.__init__(self, binary_graph_path, graph_config, calculator_params, graph_options, side_inputs, outputs, stream_type_hints)
    244 if graph_options:
    245   self._set_extension(canonical_graph_config_proto.graph_options,
    246                       graph_options)
--> 248 self._graph = calculator_graph.CalculatorGraph(
    249     graph_config=canonical_graph_config_proto)
    250 self._simulated_timestamp = 0
    251 self._graph_outputs = {}

RuntimeError: ValidatedGraphConfig Initialization failed.
ImageToTensorCalculator: ; RET_CHECK failure (mediapipe/calculators/tensor/image_to_tensor_calculator.cc:144) ValidateOptionOutputDims(options) returned INTERNAL: ; RET_CHECK failure (./mediapipe/calculators/tensor/image_to_tensor_utils.h:136) options.has_output_tensor_float_range() || options.has_output_tensor_int_range() || options.has_output_tensor_uint_range()Output tensor range is required.
InferenceCalculatorCpu: ; RET_CHECK failure (mediapipe/calculators/tensor/inference_calculator_cpu.cc:63) !options.model_path().empty() ^ kSideInModel(cc).IsConnected()Either model as side packet or model path in options is required.
SplitDetectionVectorCalculator: The number of output streams should match the number of ranges specified in the CalculatorOptions.
ImageToTensorCalculator: ; RET_CHECK failure (mediapipe/calculators/tensor/image_to_tensor_calculator.cc:144) ValidateOptionOutputDims(options) returned INTERNAL: ; RET_CHECK failure (./mediapipe/calculators/tensor/image_to_tensor_utils.h:136) options.has_output_tensor_float_range() || options.has_output_tensor_int_range() || options.has_output_tensor_uint_range()Output tensor range is required.
ConstantSidePacketCalculator: ; RET_CHECK failure (mediapipe/calculators/core/constant_side_packet_calculator.cc:64) (cc->OutputSidePackets().NumEntries(kPacketTag))==(options.packet_size())Number of output side packets has to be same as number of packets configured in options.
ConstantSidePacketCalculator: ; RET_CHECK failure (mediapipe/calculators/core/constant_side_packet_calculator.cc:64) (cc->OutputSidePackets().NumEntries(kPacketTag))==(options.packet_size())Number of output side packets has to be same as number of packets configured in options.
ConstantSidePacketCalculator: ; RET_CHECK failure (mediapipe/calculators/core/constant_side_packet_calculator.cc:64) (cc->OutputSidePackets().NumEntries(kPacketTag))==(options.packet_size())Number of output side packets has to be same as number of packets configured in options.
SplitTensorVectorCalculator: The number of output streams should match the number of ranges specified in the CalculatorOptions.
SplitNormalizedLandmarkListCalculator: The number of output streams should match the number of ranges specified in the CalculatorOptions.
SplitLandmarkListCalculator: The number of output streams should match the number of ranges specified in the CalculatorOptions.```

**Comment by ayushgdev** [comment]
Hello @rickypang0219 
From the code excerpt provided, it looks like you are using legacy/outdated version of mediapipe. We have released new MediaPipe Tasks APIs which are superset of the legacy version and much more stable. We will highly recommend you to use the newer APIs for pose landmarks since we have ended support for legacy solutions. Pose Landmarker documentation can be found [here](https://developers.google.com/mediapipe/solutions/vision/pose_landmarker/python)
